nodejs/eks/nodegroup.ts

```diff
@@ -1867,24 +1880,26 @@ export function isGravitonInstance(instanceType: string): boolean {
  * architecutre of the instance type.
  */
 function getAMIType(
-    amiType: pulumi.Input<string> | undefined,
-    gpu: pulumi.Input<boolean> | undefined,
-    instanceType: pulumi.Input<string> | undefined,
-): string {
-    if (amiType) {
-        // Return the user-specified AMI type.
-        return amiType.toString();
-    }
+    amiType: pulumi.Input<string> | pulumi.Output<string> | undefined,
+    gpu: pulumi.Input<boolean> | pulumi.Output<boolean> | undefined,
+    instanceType: pulumi.Input<string> | pulumi.Output<string> | undefined,
+): pulumi.Output<string> {
```

Comment:
This will cause failures because `amiType` is used as a literal string in line 1840

Reply:
Oof. Are there some linter settings that could've caught that or that's just on reviewers?

Reply:
Good catch! I'm not sure if a linter would be able to catch this since we're formatting an object in a string literal. The existing test cases do catch this error though, hence the failing test.